solutions/USACO jan17-balanced-photo.mdx

```diff
@@ -0,0 +1,64 @@
+---
+id: usaco-693
+source: USACO Gold 2017 January
+title: Balanced Photo
+author: Andy Wang
+---
+
+[Official Analysis](http://usaco.org/current/data/sol_bphoto_gold_jan17.html)
+
+The official solution uses [binary indexed tree](https://usaco.guide/gold/PURS/#binary-indexed-tree)
+
+An alternative solution uses order statistic tree.(C++ only)
+
+## Solution(C++ only)
+
+In order to find the cows taller than the current cow on its left and right, we maintain two balanced binary search trees storing the heights of the cows on the left and right of the current cow. Then, using function `lower_bound`, we can find the first cow in the set that is higher than the current cow. 
+
+However, STL `set` is **NOT** a random access data-structure, which means that the time complexity of finding all cows higher than the current cow is $O(N)$.(e.g using `distance()`)
+
+To lower the time complexity, we can use an [order statistic tree](https://usaco.guide/gold/PURS/#order-statistic-tree) to find the number of cows that are higher than the current one in $O(Nlog(N))$ time.
```

Comment:
Use $\mathcal O$ instead of just $O$

Reply:
Also use $\log$ instead of $log$

Reply:
change link to `/gold/PURS/#order-statistic-tree`.